feeder/util/mqtt/topic.py

```diff
@@ -0,0 +1,49 @@
+import asyncio
+import re
+
+from hbmqtt.plugins.topic_checking import BaseTopicPlugin
+from feeder.util.mqtt.authentication import local_username
+
+
+class PetnetTopicPlugin(BaseTopicPlugin):
+    feeder_sub_topic_regex = re.compile(
+        r"krs/(api|cmd)/stg/(?P<gateway_id>.*)$")
+    username_regex = re.compile(r'^/pegasus:(?P<gateway_id>.*)$')
+
+    def __init__(self, context):
+        super().__init__(context)
+
+    @asyncio.coroutine
+    async def topic_filtering(self, *args, **kwargs):
+        filter_result = super().topic_filtering(*args, **kwargs)
+        if not filter_result:
+            return False
+
+        session = kwargs.get('session', None)
+        topic = kwargs.get('topic', None)
+        self.context.logger.debug('username: %s, topic: %s' %
```

Comment:
Oh, you should also be passing your logging variables directly to the logger call along with the format string. This lets the logging module handle the substitution.

E.g. `info(“some message %s with %s”, var1, var2)`